x-pack/legacy/plugins/siem/public/components/page/network/network_top_n_flow_table/index.tsx

```diff
@@ -46,7 +46,11 @@ interface NetworkTopNFlowTableReduxProps {
 }
 
 interface NetworkTopNFlowTableDispatchProps {
-  updateTableActivePage: ActionCreator<{
+  updateIpDetailsTableActivePage: ActionCreator<{
+    activePage: number;
+    tableType: networkModel.IpDetailsTableType;
+  }>;
+  updateNetworkPageTableActivePage: ActionCreator<{
```

Comment:
Instead of having separate actions for each page, consolidating to a single action that also takes `NetworkType` should simplify the logic and make it simpler to add this component to another page in the future. Should take care of the typing issue below as well! As above the `Authentications table` has a good example of this:

https://github.com/elastic/kibana/blob/4ebcace93a16652b2c78a9a885bcef1eac7d193b/x-pack/legacy/plugins/siem/public/components/page/hosts/authentications_table/index.tsx#L49-L53

Reply:
having type issues trying to implement this. the reason we dont have the same issue in hosts is because they display the same table, where as there is a difference between network and ip details

Reply:
@spong and i paired on this a while today, thought we came up with a solution but it was a farce! Reconvening Monday